core/src/main/java/com/yahoo/oak/MemoryManager.java

```diff
@@ -46,12 +46,58 @@
     /**
      * Fetch the buffer for an allocation that is already set with its parameters: blockID, offset and length.
      */
-    void readByteBuffer(Slice s);
+    void readByteBuffer(Slice s, int blockID);
 
     /**
-     * TODO Liran: This should be documented. Why is the version handled in the memory allocator?
-     *
+     * Version is responsibility of the Memory Manager, but version shouldn't be exposed outside.
+     * This is only for testing!!
      * @return the current version
      */
     int getCurrentVersion();
+
+    /**
+     * Get ReferenceCodec to manage the (long) references,
+     * in which all the info for the memory access is incorporated
+     *
+     */
+    ReferenceCodec getReferenceCodec();
+
+    /* ------------- Interfaces to deal with references! ------------- */
+    /* Reference is a long (64 bits) that should encapsulate all the information required
+    ** to access a memory for read and for write. It is up to memory manager what to put inside.
+    ** Memory Manager should efficiently provide:
+    ** (1) Memory allocation services
+    ** (2) Memory reclamation services
+    ** (3) Safe access, in a way that a reused memory allocated to a new usage can be never accessed
+    ** via old references to the same location. */
```

Comment:
Establishing the MM role is a great idea. I think this part of the comment should be at the top of this file.